discovery-provider/src/utils/redis_cache.py

```diff
@@ -26,56 +27,69 @@ def extract_key(path, arg_items, cache_prefix_override=None):
     return key
 
 
-# NOTE: This function is deprecated in favor of `get_json_cached_key`
-# The value must also be set with `set_json_cached_key`
-def get_pickled_key(redis, key):
-    cached_value = redis.get(key)
-    if cached_value:
-        logger.debug(f"Redis Cache - hit {key}")
-        try:
-            deserialized = pickle.loads(cached_value)
-            return deserialized
-        except Exception as e:
-            logger.warning(f"Unable to deserialize cached response: {e}")
-            return None
-    logger.debug(f"Redis Cache - miss {key}")
-    return None
-
-
-# NOTE: This function is deprecated in favor of `set_json_cached_key`
-# The value must also be retrieved with `get_json_cached_key`
-def pickle_and_set(redis, key, obj, ttl=None):
-    serialized = pickle.dumps(obj)
-    redis.set(key, serialized, ttl)
-
-
 def use_redis_cache(key, ttl_sec, work_func):
     """Attempts to return value by key, otherwise caches and returns `work_func`"""
     redis = redis_connection.get_redis()
-    cached_value = get_pickled_key(redis, key)
+    cached_value = get_json_cached_key(redis, key)
     if cached_value:
         return cached_value
     to_cache = work_func()
-    pickle_and_set(redis, key, to_cache, ttl_sec)
+    set_json_cached_key(redis, key, to_cache, ttl_sec)
     return to_cache
 
 
-def get_json_cached_key(redis, key):
+def get_json_cached_key(redis, key: str) -> Any:
+    """
+    Gets a JSON serialized value from the cache.
+    """
     cached_value = redis.get(key)
     if cached_value:
         logger.debug(f"Redis Cache - hit {key}")
         try:
-            deserialized = json.loads(cached_value)
+            deserialized = json.loads(cached_value, cls=JSONDecoder)
```

Comment:
What's the impact of using the JSONDecoder from flask.json?

Reply:
I did this for consistency since we push via the flask JSONEncoder. It's not smart enough to auto parse datetime into timestamp, so i want to make sure i hit the product super hard and verify dates are correctly with QA. it actually is the same as the default: https://flask.palletsprojects.com/en/2.0.x/api/#flask.json.JSONDecoder